### PR TITLE
feat: list view of references in sidebar

### LINKED ISSUE #202: "List view" of many references in left sidebar
> "List view" of many references in left sidebar: list view with sort, filter, add, export functionality at the top

From #159 

This can be implemented like the following image:

- Add and export actions in the panel actions (right icons)
- Sort in a menu triggered by the rightmost icon (3 vertical dots)
- Filter by typing in the text area

image

@hammer how do you see the export functionality? Should we download a .bib file?

### PR DESCRIPTION
**Pull request by cguedes**
This PR fixes #202 

**Notes:**
- The REMOVE action is still not implemented (would be implemented in #198)
- This PR stop saving the references to the local storage. Instead reads references from the backend on startup
- ~**You need to RESET all references and ingest again to use this new feature**~

image

### CODE REVIEW

File: python/README.md [hunk 1]
```diff
@@ -37,6 +37,7 @@ $ poetry run pytest --cov=. tests
 The application current has three main functions:
 1. Ingest
 2. Ingest Status
+2. Ingest References
 3. Rewrite
 4. Chat
 5. Delete (References)
```
Comment (sehyod) [inline python/README.md:43]: It should say `seven main functions` now.
Also there are two number 2 in the list

Comment (cguedes) [reply python/README.md:43]: Good catch. Replaced to they auto-numbering with `1.` everywhere.

File: src/io/filesystem.ts [hunk 2]
```diff
@@ -19,7 +19,7 @@ const FILE2_CONTENT = `# Lorem Buzzword
...
...astonishing to you, that's because it is! Quick: do you have a infinitely reconfigurable scheme for coping with emerging methodologies? Is it more important for something to be dynamic or to be customer-directed? What does the buzzword 'technologies' really mean? Think virally-distributed. Our functionality is unparalleled, but our C2C2C paradigms and easy use is usually considered a terrific achievement. We think that most C2C2C web-based applications use far too much Python, and not enough Java. Have you ever been unable to disintermediate your feature set? Free? We apply the proverb 'A rolling stone gathers no moss' not only to our power to repurpose. Our functionality is unparalleled, but our power to deliver. That is a remarkable achievement taking into account this month's financial state of things! If all of this comes off as mixed-up to you, that's because it is! What does the buzzword 'technologies' really mean? Think virally-distributed. Quick: do you have a infinitely reconfigurable scheme for coping with emerging methodologies? Is it more important for something to be leading-edge or to be best-of-breed? The portals factor can be summed up in one word: affiliate-based..
```
[the remark this answers is not in the document]
Comment (cguedes) [reply src/io/filesystem.ts:22]: I've added that extra . for a "dummy" commit (GH pull requests were acting up)

File: src-tauri/Cargo.toml [hunk 3]
```diff
@@ -13,7 +13,7 @@ edition = "2021"
 tauri-build = { version = "1.2", features = [] }
 
 [dependencies]
-tauri = { version = "1.2", features = ["fs-all", "path-all", "shell-all"] }
+tauri = { version = "1.2", features = [ "fs-all", "path-all", "shell-all"] }
```
[the remark this answers is not in the document]
Comment (cguedes) [reply src-tauri/Cargo.toml:16]: It wasn't. I don't know why this changed but I will revert.

**Review by sehyod — APPROVED** [review]

**Pull request merged**